### PR TITLE
fix: add authentication to edit profile page

### LINKED ISSUE #566: [BUG] Edit profile when not logged in
## Expected Behavior

A user should not be able to access the edit page when they are not logged in.

## Actual Behavior

By going to url   ``https://earn.superteam.fun/t/<username>/edit/`` an edit page is visible to the user even when they are not logged in. 
The most important part of the issue is **it works for any username**
However, the backend logic avoids the changes so there are no worries currently but it still is a bug that needs to be handled.

## Screenshots
image

## Steps to Reproduce the Problem

1. Do not be logged in
2. Go to the URL provided above with a username
3. The page will be visible and accessed

By setting up locally, I found the issue and fixed it. It was a minor one :) 
would love to raise a PR if **you can assign this issue to me** ! 

Cheers !

### PR DESCRIPTION
**Pull request by nishaaannnt**
## What does this PR do?
Fixes #566  :  User cannot access edit page unless loggedin

## Where should the reviewer start?
Just check the line changed. It solves the issue by a single logic

## How should this be manually tested?
Try to go to the URL mentioned in the issue without being logged in. The page will redirect you to the users profile i.e with the username.

## Any background context you want to provide?
Also please just check after log in if it works correctly. (just wanna cross check)

## What are the relevant issues?
The logic was written for only if the user is logged in. 

## Screenshots (if appropriate)
Not applicable

Let me know if any changes need to be done. 
Cheers.

### CODE REVIEW

File: src/pages/t/[slug]/edit.tsx [hunk 1]
```diff
@@ -375,7 +375,7 @@ export default function EditProfilePage({ slug }: { slug: string }) {
   };
 
   useEffect(() => {
-    if (userInfo && slug !== userInfo?.username) {
+    if (!userInfo || (userInfo && slug !== userInfo?.username)) {
```
Comment (a20hek) [inline src/pages/t/[slug]/edit.tsx:378]: this does not solve the issue; better to reroute to `/` for non logged out user; plus would suggest you to check that through `useSession`

**Review by a20hek — CHANGES_REQUESTED** [review]
also, why did you delete `pnpm.lock`?

File: src/pages/t/[slug]/edit.tsx (hunk 1 above)
Comment (nishaaannnt) [reply src/pages/t/[slug]/edit.tsx:378]: ok will do. Also fixed the pnpm lock issue from my side. misunderstood the contributing guidelines.

**Comment by nishaaannnt** [comment]
@a20hek Fixed it with useSession. 
Tested in both logged In and logged out condition. Works well.
Please verify Let me know for any other change. 

Also, Error handling is not done currently in the page. i.e 
in useEffect `fetchPoW` is called without try-catch. Just pointing it out as it throws react error. Let me know if you need that fixed too or you are planning to do it yourself.

**Comment by a20hek** [comment]
> @a20hek Fixed it with useSession. Tested in both logged In and logged out condition. Works well. Please verify Let me know for any other change.
> 
> Also, Error handling is not done currently in the page. i.e in useEffect `fetchPoW` is called without try-catch. Just pointing it out as it throws react error. Let me know if you need that fixed too or you are planning to do it yourself.

you can attempt that as well

**Comment by nishaaannnt** [comment]
> you can attempt that as well

Will do but just wanted to know ifa specific toast message needs to be displayed? I guess A user Info error would be better. Let me know. or just a console.log in catch ?

**Comment by nishaaannnt** [comment]
@a20hek Do verify and let me know for any changes. Error is handled and an error message is displayed (if any) using toast.

**Review by a20hek — APPROVED** [review]
lgtm

**Pull request merged**